packages/mgt-element/src/providers/Providers.ts

```diff
@@ -112,6 +116,55 @@ export class Providers {
     return this._me;
   }
 
+  /**
+   * Gets the cache ID, creates one if it does not exist
+   *
+   * @static
+   * @memberof Providers
+   */
+  public static async getCacheId(force?: boolean) {
+    if (this._cacheId && !force) {
+      return this._cacheId;
+    }
+    if (Providers.globalProvider && Providers.globalProvider.state == ProviderState.SignedIn) {
+      if (!this._cacheId || force) {
+        this._cacheId = null;
+        const client = this.client;
+        if (client) {
+          try {
+            const response: User = await client.api('me').get();
+            if (response && response.id) {
+              this._cacheId = this.createCacheId(response);
+            }
+          } catch {}
+        }
+      }
+    }
+    return this._cacheId;
+  }
+
+  /**
+   * Unset the cache ID
+   *
+   * @static
+   * @memberof Providers
+   */
+  public static unsetCacheId() {
+    this._cacheId = null;
+  }
+
+  /**
+   * Create a cache ID
+   *
+   * @static
+   * @param {User} response
+   * @return {*}
+   * @memberof Providers
+   */
+  public static createCacheId(response: User) {
+    return response.id + '-' + response.userPrincipalName;
```

Comment:
Instead of calling `me` and using the userPrincipalName, what do you think about asking each provider to provider the account id and tenant id? - I see the account info already includes this in the Msal2Provider. Can we expand `IProviderAccount` to include tenant id? We can always fallback to `me` if a provider in the future does not provide id and tenant id